### ISSUE TITLE
`readonly` flagged invalid when preceded by `local`

### ISSUE THREAD

**Issue by zx8** [koalaman/shellcheck#1178]
#### For bugs
- SC2034 
- v0.4.7 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

function foo() {
  local readonly bar=baz
  echo "$bar"
}

```

#### Here's what shellcheck currently says:
```
In foo.sh line 4:
  local readonly bar=baz
        ^-- SC2034: readonly appears unused. Verify it or export it.
```

#### Here's what I wanted or expected to see:
Nothing.

**Comment by Nightfirecat** [comment]
`local readonly baz=bar` simply sets a local variable `baz` with value `"bar"`; the readonly word doesn't actually make the variable read-only.

That said, this warning isn't correct; no `readonly` variable is set as a result of this command.

**Comment by charlescanato** [comment]
Just perfect, Jordan.

This is one of the Bash mysteries (or bugs?). I don't understand why `readonly` is accepted after `local` and yet, it is ignored. You can't even find a reference to this possibility inside Bash documentation (at least, in the one I have locally).

And that would be good: it'd make scope and usage of the variable beautifully clear.

AFAIK, the only correct way to have a *local* and *readonly* variable is by using `declare -r baz=bar`. Yep, not so descriptive, but functional.

**Comment by koalaman** [comment]
`readonly` isn't "accepted but ignored" after `local`. You just created a local variable named `readonly` and shellcheck correctly flagged it as unused.
